### PR TITLE
LevelOfTheory attribute standardization

### PR DESCRIPTION
**Pull request by cgrambow**
### Motivation or Problem
`LevelOfTheory` standardizes all of its attributes upon initialization, which can cause issues in software like ARC that must differentiate between different string representations. See ReactionMechanismGenerator/ARC#399.

### Description of Changes
Implement  `LevelOfTheory._std_tuple`, which stores the standardized representation of the object, and no longer modify the public attributes which are provided as arguments during instance creation. `LevelOfTheory._std_tuple` is used for equality comparisons and hashing, thus retaining the functionality of using level of theory objects as dictionary keys and standard representations of levels of theory.

### Testing
Expanded unit tests to reflect the changes.

### Additional Questions
Currently, the database (`quantum_corrections/data.py`) uses the standard string representations for all of the attributes. Do we want to change those to use "nonstandard" strings for better readability? Not necessary for this PR and doesn't affect any Arkane functionality.

### CODE REVIEW

**Review by alongd — COMMENTED** [review]
Code looks great, I have no comments, added a minor question.
Re which representation to use in the database, it might not be a bad idea to use the non-standardized form. Since the data depends on the software used, it sounds reasonable to also replace it with the name the software used. But I'm OK with any choice.

File: arkane/modelchem.py [hunk 1]
```diff
@@ -50,8 +50,8 @@ def standardize_name(name: str) -> str:
         return name
 
 
-with open(os.path.join(settings['database.directory'], 'quantum_corrections', 'lot_constraints.yml')) as f:
-    METHODS_THAT_REQUIRE_SOFTWARE = yaml.safe_load(f)['METHODS_THAT_REQUIRE_SOFTWARE']
+with open(os.path.join(settings['database.directory'], 'quantum_corrections', 'lot_constraints.yml')) as _f:
```
[the remark this answers is not in the document]
Comment (cgrambow) [reply arkane/modelchem.py:53]: I use `f` in a couple of the methods later on, so I wanted to avoid having a module-level variable with the same name.

**Comment by amarkpayne** [comment]
@oscarwumit would something like this be helpful or hurtful for ARC?

**Comment by JacksonBurns** [comment]
Changing base to `main` to find out how out of date this PR is.

**Comment by JacksonBurns** [comment]
Conflicts were simple to resolve - @amarkpayne @oscarwumit @alongd @cgrambow please check this PR after tests run and see where we stand.

**Pull request closed without merging**